flutter/lib/src/flutter_enricher_event_processor.dart

```diff
@@ -0,0 +1,206 @@
+
+    return event.copyWith(
+      contexts: contexts,
+      modules: await _getPackages(),
+    );
+  }
+
+  /// Packages are loaded from [LicenseRegistry].
+  /// This is currently the only way to know which packages are used.
+  /// This however has some drawbacks:
+  /// - Only packages with licenses are known
+  /// - No version information is available
+  /// - Flutter's native dependencies are also included.
+  FutureOr<Map<String, String>> _getPackages() async {
+    if (_packages.isEmpty) {
+      // This can take some time.
+      // Therefore we cache this after running
+      var packages = <String>{};
+      // The license registry has a list of licenses entries (MIT, Apache...).
+      // Each license entry has a list of packages which licensed under this particular license.
+      // Libraries can be dual licensed.
+      //
+      // We don't care about those license issues, we just want each package name once.
+      // Therefore we add each name to a set to make sure we only add it once.
+      await LicenseRegistry.licenses.forEach(
+        (entry) => packages.addAll(
+          entry.packages.toList(),
+        ),
+      );
+
+      _packages = Map.fromEntries(
+        packages.map(
+          (e) => MapEntry(e, 'unknown'),
+        ),
+      );
+    }
+    return _packages;
+  }
+
+  Map<String, dynamic> _getCulture() {
+    // The editor says it's fine without a `?` but the compiler complains
+    // if it's missing
+    // ignore: invalid_null_aware_operator
+    final languageTag = _window.locale?.toLanguageTag();
+
+    // The editor says it's fine without a `?` but the compiler complains
+    // if it's missing
+    final availableLocales =
+        // ignore: invalid_null_aware_operator
+        _window.locales?.map((it) => it.toLanguageTag()).toList();
+
+    return <String, dynamic>{
+      'is_24_hour_format': _window.alwaysUse24HourFormat,
+      if (languageTag != null) 'locale': languageTag,
+      if (availableLocales != null) 'available_locales': availableLocales,
+      'timezone': DateTime.now().timeZoneName,
+    };
+  }
+
+  Map<String, dynamic> _getFlutterContext() {
+    final currentLifecycle = _widgetsBinding.lifecycleState;
+
+    return <String, dynamic>{
+      if (debugBrightnessOverride != null)
+        'debug_brightness_override': debugBrightnessOverride,
+      if (debugDefaultTargetPlatformOverride != null)
+        'debug_default_target_platform_override':
+            debugDefaultTargetPlatformOverride,
+      if (_window.initialLifecycleState.isNotEmpty)
+        'initial_lifecycle_state': _window.initialLifecycleState,
+      if (_window.defaultRouteName.isNotEmpty)
+        'default_route_name': _window.defaultRouteName,
+      if (currentLifecycle != null)
+        'current_lifecycle_state': describeEnum(currentLifecycle),
+      // Seems to always return false.
+      // Also always fails in tests.
+      // 'window_is_visible': _window.viewConfiguration.visible,
```

Comment:
should we do smt here?

Reply:
https://github.com/flutter/flutter/issues/83919